### ISSUE TITLE
Turn on eager-classloading by default to optimize

### ISSUE THREAD

**Issue by davsclaus** [apache/camel-quarkus#2425]
In camel-main there is a new option

camel.main.eager-classloading = true

We can be a bit smarter in camel-quarkus, and so some kind of build step, that triggers invoking this eager classloading inside camel-main. This helps to ensure those classes are loaded during build time instead of runtime at the first message processed by Camel.

We need to ensure that this is triggered as part of quarkus and its built time phase so its triggered for both JVM vs native mode

**Comment by jamesnetherton** [comment]
Can we not make `camel.main.eager-classloading = true` the default on the Camel side? Then there'd effectively be no work required in CQ given that the Camel Main bootstrapping is already done at build time.

**Comment by davsclaus** [comment]
Yeah its something that makes sense for camel-main based apps to have.

There is more work needed in core to optimize for eager loading in some of the EIPs - Have not had the time to do that.
Fell free to close this ticket.
